service/frontend.js

```diff
@@ -106,7 +106,10 @@ module.exports = class WebFrontend extends events.EventEmitter {
                 (req.protocol === 'http' && port === ':80') ||
                 (req.protocol === 'https' && port === ':443'))
                 port = '';
-            this._platform._setOrigin(req.protocol + '://' + req.hostname + port);
+            if (req.headers["x-forwarded-host"] && req.headers["x-forwarded-proto"])
+                this._platform._setOrigin(req.headers["x-forwarded-proto"] + '://' + req.headers["x-forwarded-host"]);
```

Comment:
Express already does this if we set "trust proxy" correctly (see https://expressjs.com/en/guide/behind-proxies.html ). That's controlled by `THINGENGINE_HAS_REVERSE_PROXY` in the environment.